compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPromoteMatmulToFitMMA.cpp

```diff
@@ -106,8 +97,48 @@ class LLVMGPUPromoteMatmulToFitMMAPass final
     });
 
     IRRewriter rewriter(ctx);
-    for (auto op : candidates) {
-      padWithZeroValue(rewriter, op, targetIterType, nofold);
+    for (linalg::LinalgOp op : candidates) {
+      SmallVector<int64_t> padMultiples(op.getNumLoops(), 1);
+      auto config = dyn_cast_or_null<IREE::GPU::LoweringConfigAttr>(
+          getLoweringConfig(op));
+      if (config) {
+        switch (targetDimensions) {
+        case LLVMGPUMatmulPadOption::ParallelDims:
+          padMultiples = config.getStaticTilingLevelSizes(
+              (unsigned)IREE::GPU::TilingLevel::Workgroup, op);
```

Comment:
nit: avoid the c-style cast. You can either use `llvm::to_underlying()` or `static_cast<unsigned>` in this case.